### PR TITLE
[CU-1zhbq2x] Develop main website nav popup to STE

### PR DESCRIPTION
**Pull request by vanessavmac**
## Changes
- Add popup below nav bar to direct viewers to the Sustainable Tech website
- Size popup to fit on mobile
- Delete the `node_modules` directory, `package-lock.json` or `yarn.lock` files in the outer `waterloop-website-w20` 
- Add final text

Screen Shot 2022-03-06 at 4 47 49 PM

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [x] Clicked on the popup and was redirected to /sustainable-tech
- [x] Visited each waterloop website page to ensure the deleted `node_modules` did not cause any issues

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation/READ-ME/UMLS
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by jennchenn** [comment]
Task linked: [CU-1zhbq2x Develop main website nav popup to STE](https://app.clickup.com/t/1zhbq2x)

**Comment by eevanwong** [comment]
Hey, don't forget to make this PR to the `sustainable-tech-event` branch.

**Comment by eevanwong** [comment]
Also, did you follow the package-lock.json with `npm ci`? There's alot of changes here from dependencies, which shouldn't have changed.

**Review by eevanwong — CHANGES_REQUESTED** [review]
Hey, don't forget to make this PR to the `sustainable-tech-event` branch.

Also, there seems to be alot of changes in the node_modules, did you by chance install a dependency? Could you specify which one?  I don't believe there should have been this much change in such a a single popup.

**Comment by vanessavmac** [comment]
Ok, just changed the base branch. I installed packages to use [styled icons](https://github.com/styled-icons/styled-icons). I also installed `npm install react-router-dom@5.2.0`

Screen Shot 2022-02-20 at 4 17 46 PM

**Comment by eevanwong** [comment]
Oh I see, you installed dependencies in the main directory. 

image

Yeah this is a bit confusing, and its super weird as to why whoever originally made this made it this way. To continue with your task. Can you delete node_modules, package-lock.json, and package-lock in the main directory (i.e, the directory with the waterloop-site folder)?

image

**Comment by vanessavmac** [comment]
Just made the edits! Let me know if I should make any other changes.

**Comment by eevanwong** [comment]
Yeah, I edited my request message, also please delete the circled files/folders and push those changes, we don't want node_modules, package-lock.json, package.json, or yarn.lock in the main directory, only in the waterloop-site directory.

**Comment by vanessavmac** [comment]
Ok sounds good, I deleted the files.

**Comment by eevanwong** [comment]
Looks good! Like the other PR, we shouldn't merge until we get the actual content. Once we do, we implement it and then merge it.

**Comment by vanessavmac** [comment]
Ok, thanks for the help!

File: waterloop-site/src/pages/Home.tsx
Comment (eevanwong) [inline waterloop-site/src/pages/Home.tsx:38]: Looking through some of the code I realized there were a few things I missed. Just make this a singular component `<NavPopup/>`

File: waterloop-site/package.json
Comment (eevanwong) [inline waterloop-site/package.json:9]: To stay inline with what @williamhpark was saying with your other PR, is there a way to incorporate libraries we already used (e.g material-ui) instead of installing another dependency?

File: waterloop-site/src/components/NavPopup/NavPopup.tsx
Comment (williamhpark) [inline waterloop-site/src/components/NavPopup/NavPopup.tsx:1]: Looks like this file wasn't formatted with Prettier, please reformat :)

File: waterloop-site/package.json
Comment (williamhpark) [reply waterloop-site/package.json:9]: Yep you should be able to use this: https://fonts.google.com/icons?icon.query=close with `@material-ui/icons`

File: waterloop-site/src/components/NavPopup/NavPopup.tsx
Comment (williamhpark) [inline waterloop-site/src/components/NavPopup/NavPopup.tsx:78]: Can you actually implement this functionality using React Router instead? I would suggest using `Link`.

`import { Link } from 'react-router-dom';`

File: waterloop-site/src/pages/Home.tsx
Comment (williamhpark) [inline waterloop-site/src/pages/Home.tsx:2]: Can you leave a newline between this import and the `React` import?

**Review by williamhpark — CHANGES_REQUESTED** [review]
Looks and works great! Small change requests from me :)

**Comment by vanessavmac** [comment]
Thank you @williamhpark and @eevanwong for taking a look through everything! I made all the changes except I'm not sure how to restore the `node_modules` directory, `package-lock.json` or `yarn.lock` files in the outer `waterloop-website-w20`. Would reverting 0424ab2479dcf897274168a1a379c65f34bc6ba6 do the trick?

**Comment by eevanwong** [comment]
@vanessavmac I believe that yes, revertting that would work based on solutions online. However, try to experiment to confirm so, it could have unlikely consequences that we didn't forsee. Try that solution for now and see where it goes.

**Review by williamhpark — APPROVED** [review]

File: waterloop-site/src/components/NavPopup/NavPopup.tsx
Comment (williamhpark) [inline waterloop-site/src/components/NavPopup/NavPopup.tsx:92]: I just realized that when you hover into the popup, the cursor becomes a pointer even when you just get **close** to the button but not hovering on it. This is because the `StyledLink` component wraps around the `Button`, but it should be the `Button` wrapping around the `StyledLink`, like this:
```
<Button>
        <StyledLink to="/sustainable-tech">Check it out!</StyledLink>
</Button>
```
Please make this change, then change the `color` of the `StyledLink` component to white.

**Review by williamhpark — APPROVED** [review]
One more small change I missed in my last review, but looks good besides that.

**Review by eevanwong — CHANGES_REQUESTED** [review]
image

Really quick thing, it doesn't seem intuitive to have hover over the sign up to get darker and the exit icon to get lighter when you hover over it. 

It would be better to have both be of lighter color (or opacity), then get darker when you hover over either of them.

**Review by eevanwong — APPROVED** [review]
LGTM!

**Pull request merged**
